frontend/src/common/MultiFileInput.tsx

```diff
@@ -0,0 +1,89 @@
+import { Box, Chip } from '@mui/material'
+import Button from '@mui/material/Button'
+import { styled } from '@mui/system'
+import { ChangeEvent, Dispatch, SetStateAction } from 'react'
+
+const Input = styled('input')({
+  display: 'none',
+})
+
+const displayFilename = (filename: string) => {
+  const parts = filename.split('.')
+  const ext = parts.pop()
+  const base = parts.join('.')
+  return base.length > 12 ? `${base}...${ext}` : filename
+}
+
+export default function MultiFileInput({
+  label,
+  setFiles,
+  files,
+  accepts,
+  disabled,
+  fullWidth = false,
+}: {
```

Comment:
Can this inline type be pulled out into a reusable type? I find this generally helps with readability and also allows us to easily reference the type if needed.
```
type MultiFileInputProps = {
  label: string
  setFiles: Dispatch<SetStateAction<File[] | undefined>>
  files?: File[]
  accepts?: string
  disabled?: boolean
  fullWidth?: boolean
}
```